### PR TITLE
DEV: implement SurveyActivator

### PR DESCRIPTION
**Pull request by TonisOrmisson**
the activateSurvey() method is long and does a number of things. 
I split it up to pieces in a separate class.

### CODE REVIEW

File: application/controllers/admin/surveyadmin.php [hunk 1]
```diff
@@ -782,7 +782,8 @@ public function activate($iSurveyID)
                 Survey::model()->resetCache(); // Make sure the saved values will be picked up
             }
 
-            $aResult = activateSurvey($iSurveyID);
+            $aResult = $surveyActivator->activate();
```
Comment (olleharstedt) [inline application/controllers/admin/surveyadmin.php]: Is `$aResult` ever checked?

**Comment by LouisGac** [comment]
I would have not done it this way, but a new model is still better than a helper ;-) 
Is it deeply tested?

File: application/controllers/admin/surveyadmin.php (hunk 1 above)
Comment (TonisOrmisson) [reply application/controllers/admin/surveyadmin.php]: I'm not sure about the controller. I was only aiming to replace the helper function with a class doing the same thing.

**Comment by TonisOrmisson** [comment]
Not deeply. 
only MySql also.

Any suggestions on how you'd done it?

My aim was to create a separate class since I have a plan of testing a version with response data stored all in a  JSON field rather than table. So I can extend & customize the activator more simply.

**Comment by TonisOrmisson** [comment]
depends on #925

**Comment by TonisOrmisson** [comment]
Ok removed the dependency to #925 and created automated tests

**Comment by olleharstedt** [comment]
I don't understand why we now, in the master branch, have both SurveyActivator *and* the old activateSurvey function?

The SurveyActivator is a good step towards putting business logic in service classes. I've made a small workshop about it in the office, you can read the PDF slides if if interested:
[cleancode_workshop.pdf](https://github.com/LimeSurvey/LimeSurvey/files/4319083/cleancode_workshop.pdf)

**Comment by Shnoulle** [comment]
activateSurvey seems deleted/removed in this commit 

https://github.com/LimeSurvey/LimeSurvey/pull/923/files#diff-da74319d41b9ca5e8e86f9a11891ff6bL255

**Comment by olleharstedt** [comment]
Yeah. So why is it still in the code-base? oO

**Comment by Shnoulle** [comment]
And added 8 years ago ....

**Comment by Shnoulle** [comment]
Removed in develop ? Merge issue develop to master ?????

**Comment by olleharstedt** [comment]
Maybe. This event was never added to SurveyActivator, it seems: "afterSurveyActivate".

**Comment by olleharstedt** [comment]
Fixed.

**Pull request merged**